numpy/core/tests/test_regression.py

```diff
@@ -2115,6 +2115,10 @@ def __array__(self,*args,**kwargs):
         for f in [op.lt, op.le, op.gt, op.ge]:
             if sys.version_info[0] >= 3:
                 assert_raises(TypeError, f, lhs, rhs)
+            elif sys.version_info[0] < 3 and sys.py3kwarning:
+                warnings.filterwarnings('error',category=DeprecationWarning)
+                assert_raises(DeprecationWarning, f, lhs, rhs)
+                warnings.filterwarnings('ignore',category=DeprecationWarning)
```

Comment:
The test is failing due to this. Don't use "ignore" filters, they are horribly broken in versions before python 3.4.x if you want to reliably test them. Also filters should always be inside a context (either catch_warnings, or our own suppress_warnings context).

My guess is: put this test into a warnings context. Then add a new test if you want to check whether it actually gives the warning when py3kwarning is active.